hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/ipc/Server.java

```diff
@@ -2516,13 +2509,15 @@ public int readAndProcess() throws IOException, InterruptedException {
           }
 
           if (!RpcConstants.HEADER.equals(dataLengthBuffer)) {
+            final String hostName = addr == null ? this.hostAddress : addr.getHostName();
```

Comment:
In what case addr would be null? We were not handling `addr == null` edge case before. Wondering what has changed.